### PR TITLE
Fix job metrics

### LINKED ISSUE #7772: Job events are counted also on followers
**Describe the bug**

In a benchmark with a process model with one service task, the metrics report 200 instances are completed but 600 jobs are completed instead of 200 jobs.

image

This happens because these events are counted on followers also.

**To Reproduce**

Check CW 35 benchmark cluster

**Expected behavior**

Events are counted only on leader.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description
EventAppliers are executed on replay and  cause counting metrics multiple times, which can lead to confusion. This PR moves the metric counting outside of the state changes into the processors.

 * fix: write job complete metric only in processor 
 * fix: write fail metrics in processor
* fix: write error metrics in processor
* fix: write time out metrics in processor
* fix: write activated metrics in processor
* fix: write cancel metrics in processor
* fix: write create metrics in processor 

The creation of the jobs are done in a BpmnBehavior which is unrelated to the job processors, this means we need to create the metrics class on a much higher level, and have to pass it through several layers which is not nice but the simplest solution for now. 

Alternative is to create a singleton, but I prefer not to do that, since we have then other issues. **OR** split it up in several metrics (currently it is one metric with several labels), such that we can have multiple metric classes. This would mean we could have one class for the created job metric, which can be used in the behavior. This would include more work and fixing the grafana panels, which I'm not sure whether this is currently necessary or worth the effort.

## Related issues

closes https://github.com/camunda-cloud/zeebe/issues/7772

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java [hunk 1]
```diff
@@ -55,6 +61,7 @@ public void createNewJob(
 
     final var jobKey = keyGenerator.nextKey();
     stateWriter.appendFollowUpEvent(jobKey, JobIntent.CREATED, jobRecord);
+    jobMetrics.jobCreated(jobType);
```
Comment (deepthidevaki) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java:64]: Do we need a metric for job creation? We have another metrics named `element_instance_events_total` in `ProcessEngineMetrics`. If I remember correctly, we can filter it by "type = SERVICE_TASK" and "action=activated". I was wondering if that is enough for job creation metrics. Service task activation corresponds to job creation, isn't it?

**Review by deepthidevaki — COMMENTED** [review]
Looks good :+1: I am wondering if we can remove job creation metric. Please see my comment.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java (hunk 1 above)
Comment (Zelldon) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java:64]: The issue is then that we need to adjust the query, which I thing is not that easy if you want to have it in one table.

query

Comment (deepthidevaki) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java:64]: Ok. There might be some way to merge multiple queries, right? But let's keep it simple and have explicit job created metrics. :+1:

**Review by deepthidevaki — APPROVED** [review]
:rocket:

**Comment by Zelldon** [comment]
Bors r+

**Comment by Zelldon** [comment]
Bors r+

**Comment by Zelldon** [comment]
Bors r+ :skull:

**Comment by Zelldon** [comment]
Bors r+

**Comment by Zelldon** [comment]
Bors r+

**Pull request merged**